### PR TITLE
QOL: Auto locate model weights based on subdirectory and epoch

### PR DESCRIPTION
**Pull request by robertklee**
This was branched from Julian's branch and contains something already in main, so there's a conflict.

It mainly adds normalization of the input image, as well as auto-locating the model weights and architecture json based on the subdirectory and epoch number.

### CODE REVIEW

File: evaluation.py [hunk 1]
```diff
@@ -29,13 +37,22 @@ def _vstack_images(self, img_list, interpolation=cv2.INTER_CUBIC):
...
-        return predict_heatmaps
+
+        print('model prediction metrics: ')
+        predict_mean = np.mean(predict_heatmaps)
+        predict_max = np.max(predict_heatmaps)
+        predict_min = np.min(predict_heatmaps)
+        predict_var = np.var(predict_heatmaps.flatten())
+        print('Mean: {:0.6e}\t Max: {:e}\t Min: {:e}\t Variance: {:e}'.format(predict_mean, predict_max, predict_min, predict_var))
+        normalized_heatmaps = predict_heatmaps / predict_max
+        normalized_heatmaps = normalized_heatmaps - predict_min
```
Comment (julianrocha) [inline evaluation.py:51]: These 2 lines get overwritten by the next

Comment (robertklee) [reply evaluation.py:51]: good catch, my bad

File: evaluation.py [hunk 2]
```diff
@@ -36,7 +40,18 @@ def predict_heatmaps(self, X):
...
+        normalized_heatmaps = predict_heatmaps / np.max(predict_heatmaps)
```
Comment (robertklee) [inline evaluation.py]:
```suggestion
        normalized_heatmaps = normalized_heatmaps / np.max(predict_heatmaps)
```

File: evaluation.py [hunk 3]
```diff
@@ -36,7 +40,18 @@ def predict_heatmaps(self, X):
...
+
+        print('model prediction metrics: ')
+        predict_mean = np.mean(predict_heatmaps)
+        predict_max = np.max(predict_heatmaps)
+        predict_min = np.min(predict_heatmaps)
+        predict_var = np.var(predict_heatmaps.flatten())
+        print('Mean: {:0.6e}\t Max: {:e}\t Min: {:e}\t Variance: {:e}'.format(predict_mean, predict_max, predict_min, predict_var))
+        normalized_heatmaps = predict_heatmaps / predict_max
+        normalized_heatmaps = normalized_heatmaps - predict_min
+        normalized_heatmaps = normalized_heatmaps / np.max(predict_heatmaps)
```
Comment (robertklee) [inline evaluation.py]:
```suggestion
        normalized_heatmaps = normalized_heatmaps / np.max(normalized_heatmaps)
```

**Pull request merged**